Neos.Fusion/Resources/Private/Fusion/Root.fusion

```diff
@@ -167,6 +167,38 @@ prototype(Neos.Fusion:ResourceUri) {
   @exceptionHandler = 'Neos\\Fusion\\Core\\ExceptionHandlers\\AbsorbingHandler'
 }
 
+# Renders a link pointing to a controller/action
+#
+# Usage:
+# link = Neos.Fusion:Link.Action {
+#   href {
+#     package = 'Some.Package'
+#     controller = 'Standard'
+#     action = 'index'
+#   }
+# }
+#
+prototype(Neos.Fusion:Link.Action) < prototype(Neos.Fusion:Tag) {
+    tagName = 'a'
+    href = Neos.Fusion:UriBuilder
+    attributes.href = ${this.href}
+}
+
+# Renders a link pointing to a resource
+#
+# Usage:
+# link = Neos.Fusion:Link.Resource {
+#   href {
+#     path = 'resource://Some.Package/Public/Images/SomeImage.png'
+#   }
+# }
+#
+prototype(Neos.Fusion:Link.Resource) < prototype(Neos.Fusion:Tag) {
+    tagName = 'a'
+    href = Neos.Fusion:ResourceUri
+    attributes.href = ${this.href}
```

Comment:
This won't work, because attributes have no access to local variables.

Something like this should work: 

```
prototype(Neos.Fusion:Link.Resource) < prototype(Neos.Fusion:Component) {
    tagName = 'a'
    href = Neos.Fusion:ResourceUri
    content = null

    renderer = Neos.Fusion:Tag {
        tagName = 'a'
        attributes {
            @ignoreProperties=${['content', 'tagName', 'content']}
            @apply.spread = ${props}
        }
        content = ${props.content}
    }
}
```

Reply:
Like that, you could write markup like this: 
```
<Neos.Fusion:Link.Resource
  href.path='resource://Some.Package/Public/Images/SomeImage.png'
  class='my-class'
>
  Content
</Neos.Fusion:Link.Resource>
```

Reply:
But I would prefer to use an eel helper:
```
<a
  href={StaticResource.uri('Some.Package', 'Public/Images/SomeImage.png')} 
  class='my-class'
>
  Content
</a>
```

Reply:
Perhaps it would be helpful to have an Eel helper pendant for `Neos.Fusion:UriBuilder`

Reply:
Hm, indeed that would be an alternative and probably better as this `Link` component just wraps the href. Thanks for the heads-up!

Reply:
I would prefer them to be components and have all input vars directly available without having to know the inner workings of `href`. Wdyt?

Reply:
That was my initial idea, but then I thought it is a was a waste of code lines to just forward all attributes to an UriBuilder. Also, like this you could just put a normal string URL into the href and it works. So, the href is just more defined in how it can be made up.

Reply:
Still think the API should be on the first level and no further knowledge of further level should be necessary. 
Would also make it easier if we have some kind of preliminary autocomplete at some point.

Reply:
Okay, I had another short look at this. I still find the Helper more attractive, as it just encompasses the generation of an Uri into a utility method which is exactly what this change is about - but:
the helper somehow needs access to the UriBuilder, meaning access to the ControllerContext, meaning access to the current ActionRequest. The Fusion Runtime provides that already (or builds one itself), but that is not available in the Eel Helper right now as far as I see, but we do have the current `request` (https://github.com/neos/neos-development-collection/blob/master/Neos.Fusion/Classes/Core/Runtime.php#L892).
So either we also pass the UriBuilder directly down into the default Eel context, or create a new one on demand from the `request`. The ResourceManager and I18nService for the ResourceUriHelper is straightforward.

Now the alternative is to make the Link.Action and Link.Resource objects fully copy the properties of UriBuilder/ResourceUri objects, means we need to keep the two pairs in sync in the future. I think I'll give that approach a try

Are there any other cases where one would want to render an action/resource URI into a tag attribute?

Reply:
I agree to @Sebobo on this one. The `href` api is not very transparent and I would expect s.th. like this instead:
```
prototype(Neos.Fusion:Link.Action) < prototype(Neos.Fusion:Component) {
    package = ''
    controller = ''
    action = ''
    # ....
    renderer = Neos.Fusion:Tag {
        tagName = 'a'
        href = Neos.Fusion:UriBuilder {
            @apply.props = ${props}
        }
    }
}
```

and a similar one for resources

> means we need to keep the two pairs in sync in the future.

I don't think/hope that they will change a lot in the future. And if so, it doesn't necessarily mean that the `Link.Action` has all features of the "lower level" prototype (that you could still use individually)

Reply:
btw: Not exactly related but regarding Eel Helper vs Fusion prototype: I think we should make it possible to use one from the other somehow. And/or get rid of the nasty `ProtectedContextAwareInterface` as hard requirement for Eel helpers. Instead it should be possible to use e.g. the `UriBuilder` directly somehow

Reply:
Please check out my attempt at an implementation everyone. I have no real clue what I'm doing :D